seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/hive/HiveTypeMapper.java

```diff
@@ -0,0 +1,120 @@
+
+    // Numeric Types
+    private static final String HIVE_TINYINT = "TINYINT";
+    private static final String HIVE_SMALLINT = "SMALLINT";
+    private static final String HIVE_INT = "INT";
+    private static final String HIVE_INTEGER = "INTEGER";
+    private static final String HIVE_BIGINT = "BIGINT";
+    private static final String HIVE_FLOAT = "FLOAT";
+    private static final String HIVE_DOUBLE = "DOUBLE";
+    private static final String HIVE_DOUBLE_PRECISION = "DOUBLE PRECISION";
+    private static final String HIVE_DECIMAL = "DECIMAL";
+    private static final String HIVE_NUMERIC = "NUMERIC";
+    // Date/Time Types
+    private static final String HIVE_TIMESTAMP = "TIMESTAMP";
+    private static final String HIVE_DATE = "DATE";
+    private static final String HIVE_INTERVAL = "INTERVAL";
+    // String Types
+    private static final String HIVE_STRING = "STRING";
+    private static final String HIVE_VARCHAR = "VARCHAR";
+    private static final String HIVE_CHAR = "CHAR";
+    // Misc Types
+    private static final String HIVE_BOOLEAN = "BOOLEAN";
+    private static final String HIVE_BINARY = "BINARY";
+    // Complex Types
+    private static final String HIVE_ARRAY = "ARRAY";
+    private static final String HIVE_MAP = "MAP";
+    // float
+    private static final String HIVE_STRUCT = "STRUCT";
+    private static final String HIVE_UNIONTYPE = "UNIONTYPE";
+
+    @Override
+    public SeaTunnelDataType<?> mapping(ResultSetMetaData metadata, int colIndex)
+            throws SQLException {
+        String columnType = metadata.getColumnTypeName(colIndex).toUpperCase();
+        int precision = metadata.getPrecision(colIndex);
+        switch (columnType) {
+            case HIVE_TINYINT:
+            case HIVE_SMALLINT:
+                return BasicType.SHORT_TYPE;
+            case HIVE_INT:
+            case HIVE_INTEGER:
+                return BasicType.INT_TYPE;
+            case HIVE_BIGINT:
+                return BasicType.LONG_TYPE;
+            case HIVE_FLOAT:
+                return BasicType.FLOAT_TYPE;
+            case HIVE_DOUBLE:
+            case HIVE_DOUBLE_PRECISION:
+                return BasicType.DOUBLE_TYPE;
+            case HIVE_DECIMAL:
+            case HIVE_NUMERIC:
+                if (precision > 0) {
+                    return new DecimalType(precision, metadata.getScale(colIndex));
+                }
+                LOG.warn("decimal did define precision,scale, will be Decimal(38,18)");
+                return new DecimalType(38, 18);
+            case HIVE_TIMESTAMP:
+                return LocalTimeType.LOCAL_DATE_TIME_TYPE;
+            case HIVE_DATE:
+                return LocalTimeType.LOCAL_DATE_TYPE;
+            case HIVE_STRING:
+            case HIVE_VARCHAR:
+            case HIVE_CHAR:
+                return BasicType.STRING_TYPE;
+            case HIVE_BOOLEAN:
+                return BasicType.BOOLEAN_TYPE;
+            case HIVE_BINARY:
```

Comment:
How about map `HIVE_BINARY ` to  `PrimitiveByteArrayType.INSTANCE`?

Reply:
don't support 
`Caused by: java.sql.SQLFeatureNotSupportedException: Method not supported
	at org.apache.hive.jdbc.HiveBaseResultSet.getBytes(HiveBaseResultSet.java:222)
	at org.apache.seatunnel.connectors.seatunnel.jdbc.internal.converter.AbstractJdbcRowConverter.toInternal(AbstractJdbcRowConverter.java:97)
	at org.apache.seatunnel.connectors.seatunnel.jdbc.internal.JdbcInputFormat.nextRecord(JdbcInputFormat.java:220)
	... 6 more`

`CREATE TABLE all_data_types
(
    int_column              INT,
    integer_column          INTEGER,
    bigint_column           BIGINT,
    smallint_column         SMALLINT,
    tinyint_column          TINYINT,
    double_column           DOUBLE,
    double_PRECISION_column DOUBLE PRECISION,
    float_column            FLOAT,
    string_column           STRING,
    char_column             CHAR(10),
    varchar_column          VARCHAR(20),
    boolean_column          BOOLEAN,
    date_column             DATE,
    timestamp_column        TIMESTAMP,
    binary_column           BINARY,
    decimal_column          DECIMAL(10, 2),
    numeric_column          NUMERIC(10, 2)
);`
`INSERT INTO all_data_types
VALUES (2,
        1,
        1234567890,
        32767,
        127,
        123.45,
        123.45,
        67.89,
        'Hello, Hive',
        'CharCol',
        'VarcharCol',
        TRUE,
        '2023-09-04',
        '2023-09-04 10:30:00',
        UNHEX('48656C6C6F2C2048697665'),
        42.12,
        42.12)`


but it can map HIVE_BINARY  to  STRING_TYPE.

Reply:
Get it.